source/Paths.cpp

```diff
@@ -36,7 +36,7 @@ struct PathException: std::runtime_error {
 		std::runtime_error(message) {
 	}
 };
-#if BOOST_OS_LINUX != 0
+#if BOOST_OS_LINUX != 0 || BOOST_OS_MACOS != 0
```

Comment:
Is this causing this block to take effect on macOS? The function contains code that reads /proc; there is no /proc on macOS.

Reply:
Of course, you are correct.
Sorry for my carelessness.
I have update the PR to use [_NSGetExecutablePath](https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/dyld.3.html).